### ISSUE TITLE
Consider writing a contributer's guide

### ISSUE THREAD

**Issue by deeplook** [jupyter-widgets/ipyleaflet#350]
I'm often stumbling over missing features in ipyleaflet that I believe would be easy to implement for someone with more of a Javascript background, and/or are not tackled for a substantial period of time, while sometimes reported by multiple people, due to different priorities. When I look at the code I can see it's usually not much for any given feature, but without being a leaflet pro I have a hard time to see the "grand architecture" and/or estimate the work needed.

Therefore I wonder if a "contributer's guide" explaining the architecture and outlining strategies for implementing new features would not maybe help accelerate contributions significantly?

**Comment by martinRenou** [comment]
I agree with you. It is a bit unclear to me where this documentation should be put, as it would be the same across most of the interactive widgets libraries. I guess we can put it in the `https://github.com/jupyter-widgets/widget-cookiecutter` project, and then put it in the documentation of ipyleaflet as well.

**Comment by deeplook** [comment]
The existing docs are mostly a gallery of example layers/controls and generating code, great to get started and copy/paste initial snippets to support one's own use-case. I do it quite often. The questions answered there so far are:

- What can I do with the package?
- What is a simple example for some layer/control?

What I think is increasingly needed is something that addresses questions like these ones:

- How do I wrap another existing leaflet "widget" in ipyleaflet for myself?
- How do I wrap one such that it can go into the package?
- How do I write a leaflet widget from scratch in a most ipyleaflet-ready way?
- How do I build something properly (javascript tooling)?
- How do I test it properly (javascript tooling)?
- What would be desired widgets to be contributed?
- What else would be desired contributions?
- What is not desired and why (policies)?

This would be helpful for example as simple things as creating a control widget on the map that simply hides or deletes a set of layers of some type (something that the current LayerControl cannot do).
